pkg/migration-tool/writer/writer.go

```diff
@@ -59,6 +58,9 @@ func New(config Config) (*Write, error) {
 			Labels: utils.LabelSet(config.ProgressMetricName, config.MigrationJobName),
 		}
 	}
+	if config.GarbageCollectOnPush {
+		write.sigGC = make(chan struct{})
```

Comment:
what's the length of this chan? Will this action be sync or async?

Reply:
This would be async since GC can be a blocking call. We don't need the length here since `isGCInProgress` boolean rejects the signal in case GC is already running, so indirectly async.

Reply:
This is NOT async right now since the channel length is 0. and the `gc` function will block  at `runtime.GC()` before taking things off the queue at `for range collect {`. You either need to run two `gc` goroutines or add slack to the channel length here

Reply:
Oh yes, thank you for pointing it out. Making the length as `2` should do the job.